### PR TITLE
Small cleanup and updates:

### PR DESCRIPTION
**Pull request by florina-muntenescu**
* Use the 1.1.1 version of Architecture Components
* Use the InstantTaskExecutorRule from the testing artifact
* Remove irrelevant menu from MainActivity
* Make the list from WordListAdapter non nullable and initially empty
* Added link to codelab in Readme
* Typo in readme fixed
* Added a comment explaining how to populate the database only when it's created for the first time

### CODE REVIEW

File: app/src/main/java/com/example/android/roomwordssample/WordListAdapter.java [hunk 1]
```diff
@@ -63,9 +65,7 @@ void setWords(List<Word> words){
     // mWords has not been updated (means initially, it's null, and we can't return null).
```
Comment (aleksinthecloud) [inline app/src/main/java/com/example/android/roomwordssample/WordListAdapter.java]: Remove comment as no longer applies.

File: app/src/main/java/com/example/android/roomwordssample/WordRoomDatabase.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
 
     public abstract WordDao wordDao();
 
-    private static WordRoomDatabase INSTANCE;
+    private static volatile WordRoomDatabase INSTANCE;
```
Comment (aleksinthecloud) [inline app/src/main/java/com/example/android/roomwordssample/WordRoomDatabase.java]: Add comment as to what "volatile" means.

File: app/src/main/java/com/example/android/roomwordssample/WordListAdapter.java (hunk 1 above)
Comment (florina-muntenescu) [reply app/src/main/java/com/example/android/roomwordssample/WordListAdapter.java]: done

File: app/src/main/java/com/example/android/roomwordssample/WordRoomDatabase.java (hunk 2 above)
Comment (florina-muntenescu) [reply app/src/main/java/com/example/android/roomwordssample/WordRoomDatabase.java]: done

**Pull request merged**